pkg/provider/imports/resource.go

```diff
@@ -0,0 +1,21 @@
+package imports
+
+import "github.com/klothoplatform/klotho/pkg/core"
+
+// Imported is an internal resource to signal that the resource that depends on this
+// should be imported from `ID`.
+type Imported struct {
+	ID string
+}
+
+func (imp Imported) KlothoConstructRef() []core.AnnotationKey {
+	return nil
+}
+
+func (imp Imported) Id() core.ResourceId {
+	return core.ResourceId{
+		Provider: "internal",
```

Comment:
The resources the import points to _can_ be aws, but isn't necessarily. The only aws-specific part about imports is in the IaC rendering.

Reply:
We do now use the provider in the visualizer's topology.yaml, fwiw (as of #573). Since the node ids can be optionally fully-qualified with a provider name, we add it iff the node's provider is different from the topo's default one (which for now we always hard-code to `aws`).

I don't know that that's a problem, but I figured I'd mention it just in case.